src/lsqecc/simulation/logical_patch_state_simulation.py

```diff
@@ -184,14 +186,129 @@ def _get_all_operating_patches(
         return patch_list
 
 
+class SimulatorType(enum.Enum):
+    FULL_STATE_VECTOR = "FullStateVector"
+    LAZY_TENSOR = "LazyTensor"
+
+
 class PatchSimulator:
     def __init__(self, logical_computation: llops.LogicalLatticeComputation):
-        # TODO decouple the simulation frm the logical computation. It should be possible to do it
-        # because the logical computation is only used to construct the mapper and the initial
-        # states.
-
         self.logical_computation = logical_computation
         self.mapper = PatchToQubitMapper(logical_computation)
+
+    def apply_logical_operation(self, logical_op: llops.LogicalLatticeOperation):
+        raise NotImplementedError
+
+    def get_separable_states(self) -> Dict[uuid.UUID, qkop.DictStateFn]:
+        raise NotImplementedError
+
+    @staticmethod
+    def make_simulator(
+        simulator_type: SimulatorType, logical_computation: llops.LogicalLatticeComputation
+    ):
+        if simulator_type == SimulatorType.LAZY_TENSOR:
+            return LazyTensorPatchSimulator(logical_computation)
+        else:
+            return FullStateVectorPatchSimulator(logical_computation)
+
+
+class FullStateVectorPatchSimulator(PatchSimulator):
```

Comment:
This is just adding back the existing state vector simulator, unfortnuately it didn't occur to me that we would probably want to keep both in the end